src/mito2/src/sst/index/fulltext_index/creator.rs

```diff
@@ -277,6 +315,41 @@ impl SingleCreator {
         Ok(())
     }
 
+    async fn update_flat(&mut self, batch: &RecordBatch) -> Result<()> {
+        // Find the column in the RecordBatch by name
+        if let Some(column_array) = batch.column_by_name(&self.column_name) {
+            // Convert Arrow array to VectorRef
+            let vector = Helper::try_into_vector(column_array.clone())
+                .context(crate::error::ConvertVectorSnafu)?;
+            // Cast to string if needed
+            let string_vector =
+                vector
+                    .cast(&ConcreteDataType::string_datatype())
+                    .context(CastVectorSnafu {
+                        from: vector.data_type(),
+                        to: ConcreteDataType::string_datatype(),
+                    })?;
+
+            for i in 0..batch.num_rows() {
+                let data = string_vector.get_ref(i);
```

Comment:
maybe downcast `string_vector` to concrete type `StringVector` and use it's `get_data` method could be more performant?